Build In-App Notifications System

robin-macpherson opened the pull request:
## Description

**Issues:** 
- closes #380 
- closes #437
- closes #191 
- closes #429 
- closes #619 

**[Approach Document](https://github.com/Journaly/journaly/blob/master/docs/approach-docs/in-app-notifications.md)** 

This work builds an entire notifications system:
- UI and UX for creating a seamless and intuitive notification experience within the app
- DB migration
- Internal infrastructure for handling this new system
- Deep Linking for feedback threads FINALLY 🌟

With this work I think we can finally close #437 and #191  🙌🏼

## Subtasks

- [x] I have added this PR to the Journaly Kanban project ✅
- [x] Data Modeling & Migrations
- [x] Resolvers
- [x] GraphQL
- [x] Mobile UI
- [x] Desktop UI
- [x] Wire up Client & Server-side
- [x] Create Notification Context & hook everything up
- [x] Create `UserList` Component and also use this for post claps
- [x] Implement Optimistic UI Behavior
  - [x] Delete
  - [x] Mark As Read
- [x] Deep Linking For Thread & Post Comments
- [x] Deep Linking For Emails
- [x] ~Browser notifications~ (moved to follow up)
- [x] Move over to `nexus-prisma` plugin?
- [x] Translations for English & German
- [x] Update Seed data
    - [x] One user empty state
    - [x] Others have some
    - [x] Jon has every notification
- [x] Create a `SwipeableElement` Component with swipey-swipe gestures
- [x] Upgrade Prisma to latest
- [x] Check all in-code TODOs

### Deployment Checklist

- [x] 🚨 Change Polling Interval back to 2 / 5 mins
- [x] 🚨 Publish new j-db-client version and update in both `web` and `j-mail`
- [x] 🚨 Deploy migs to stage
- [x] 🚨 Deploy code to stage
- [x] 🚨 DEPLOY `j-mail` to stage
- [x] 🚨 Deploy migs to prod
- [ ] 🚨 Deploy code to prod
- [ ] 🚨 DEPLOY `j-mail` TO PROD

## Migrations

- Does the migration need to be applied before or after deploying code? Yes ✅
- Is applying the migration going to necessitate downtime? No ❎
- Is there any SQL or backfill logic that has to be run manually? Yes ✅
- Are the DB changes high risk in your estimation? Should a manual DB snapshot be taken before applying? TBD

## Screenshots

#### 1. Desktop - Feed Open

<img width="1440" alt="Screenshot 2021-11-30 at 14 17 24" src="https://user-images.githubusercontent.com/34203886/144064443-f67a637f-2217-45f8-a0a6-51cdc126b25b.png">

#### 2. Mobile - Feed Open

<img width="333" alt="Screenshot 2021-11-30 at 14 17 49" src="https://user-images.githubusercontent.com/34203886/144064463-c0ca266d-7d64-406d-9489-33dace6a9f72.png">

#### 3. Mobile - Swipey-swipe actions

<img width="318" alt="Screenshot 2021-11-30 at 14 18 30" src="https://user-images.githubusercontent.com/34203886/144064486-806556ad-2e0d-47c9-b7ae-6597888eae38.png">

#### 4. Mobile - Read vs. Unread

<img width="324" alt="Screenshot 2021-11-30 at 14 18 42" src="https://user-images.githubusercontent.com/34203886/144064503-48546808-3070-4f38-b355-9709f07eaf7f.png">

robin-macpherson commented:
🤩 @Lanny iiiiiiit's **GO TIME**! 🏎💨🔥🔥🔥🔥🔥🔥

robin-macpherson commented:
We're getting there!

https://media.giphy.com/media/idFxmiV2dayJEqzXaW/giphy.gif

robin-macpherson commented:
We'reeeeee getting there!!

<img src="https://media.giphy.com/media/idFxmiV2dayJEqzXaW/giphy.gif" />

LindasLinguas commented on packages/web/public/static/locales/de/notifications.json:
```diff
@@ -0,0 +1,21 @@
+    "newFollowers": "You got {{followerCount}} new follower",
+    "threadComments": "You got {{commentCount}} new feedback comments on this post",
+    "postComments": "You got {{commentCount}} new general comments on this post",
+    "postClaps": "{{clapCount}} people clapped for your post",
```
```suggestion
    "postClaps": "{{clapCount}} Leute haben diesem Beitrag applaudiert",
```

LindasLinguas commented on packages/web/public/static/locales/de/notifications.json:
```diff
@@ -0,0 +1,21 @@
+    "threadComments": "You got {{commentCount}} new feedback comments on this post",
+    "postComments": "You got {{commentCount}} new general comments on this post",
+    "postClaps": "{{clapCount}} people clapped for your post",
+    "threadCommentThanks": "{{userIdentifier}} said thanks for {{thanksCount}} feedback comments",
```
```suggestion
    "threadCommentThanks": "{{userIdentifier}} hat für {{thanksCount}} Feedback-Kommentare Danke gesagt",
```

LindasLinguas replied:
Plural issue ...

LindasLinguas commented on packages/web/public/static/locales/de/notifications.json:
```diff
@@ -0,0 +1,21 @@
+    "postComments": "You got {{commentCount}} new general comments on this post",
+    "postClaps": "{{clapCount}} people clapped for your post",
+    "threadCommentThanks": "{{userIdentifier}} said thanks for {{thanksCount}} feedback comments",
+    "newPosts": "Journalers you follow published {{postCount}} new posts"
```
```suggestion
    "newPosts": "Journaler, denen du folgst, haben {{postCount}} neue Beiträge veröffentlicht"
```

LindasLinguas replied:
plural issue...

LindasLinguas commented on packages/web/public/static/locales/de/notifications.json:
```diff
@@ -0,0 +1,21 @@
+    "newPosts": "Journalers you follow published {{postCount}} new posts"
+  },
+  "levelTwo": {
+    "newFollowers": "These {{followerCount}} awesome Journalers are now following you",
```
```suggestion
    "newFollowers": "Diese {{followerCount}} großartigen Journaler folgen dir nun",
```

LindasLinguas replied:
plural issue...

LindasLinguas commented on packages/web/public/static/locales/de/notifications.json:
```diff
@@ -0,0 +1,21 @@
+  },
+  "levelTwo": {
+    "newFollowers": "These {{followerCount}} awesome Journalers are now following you",
+    "threadCommentThanks": "You received {{thanksCount}} thanks for feedback on this post",
```
```suggestion
    "threadCommentThanks": "Du hast {{thanksCount}} Danke für Feedback zu diesem Beitrag erhalten",
```

LindasLinguas commented on packages/web/public/static/locales/de/notifications.json:
```diff
@@ -0,0 +1,21 @@
+  "levelTwo": {
+    "newFollowers": "These {{followerCount}} awesome Journalers are now following you",
+    "threadCommentThanks": "You received {{thanksCount}} thanks for feedback on this post",
+    "postClaps": "These {{userCount}} people clapped for your post"
```
```suggestion
    "postClaps": "Diese {{userCount}} Leute haben deinem Beitrag applaudiert"
```

LindasLinguas replied:
Plural issue: Leute and people can't be a single person. 

1. This person clapped for your post = Diese Person hat deinem Beitrag applaudiert
2. These 3 people clapped for your post = Diese 3 Leute haben deinem Beitrag applaudiert

LindasLinguas commented on packages/web/public/static/locales/de/notifications.json:
```diff
@@ -0,0 +1,21 @@
+    "postClaps": "These {{userCount}} people clapped for your post"
+  },
+  "actions": {
+    "markAsRead": "Read",
```
```suggestion
    "markAsRead": "Gelesen",
```

LindasLinguas commented on packages/web/public/static/locales/de/notifications.json:
```diff
@@ -0,0 +1,21 @@
+  },
+  "actions": {
+    "markAsRead": "Read",
+    "delete": "Delete"
```
```suggestion
    "delete": "Löschen"
```

LindasLinguas reviewed (commented):
Hi Robin, 
One major thing I found is that many notification texts don't work if the count is 1. That's an issue for English **and** German. We should consider adding an extra line for when the counter is one.

robin-macpherson commented on packages/web/public/static/locales/de/notifications.json:
```diff
@@ -0,0 +1,21 @@
+  "feedTitle": "Notifications",
+  "emptyFeed": "You have no notifications... yet!",
+  "levelOne": {
+    "newFollowers": "You got {{followerCount}} new follower",
```
Thanks for including both as a comment here! Yes when I originally coded this without translations I handled both cases programmatically, but with the translation mechanism isn't a little more tricky, I'll take care of that over the weekend and include both "Du hast 1 neuen Follower.", but "Du hast 3 neue Follower."

robin-macpherson commented on packages/web/public/static/locales/de/notifications.json:
```diff
@@ -0,0 +1,22 @@
+    "newFollowers": "Diese {{followerCount}} großartigen Journaler folgen dir nun",
+    "threadCommentThanks": "Du hast {{thanksCount}} Danke für Feedback zu diesem Beitrag erhalten",
+    "postClaps": "Diese {{userCount}} Leute haben deinem Beitrag applaudiert",
+    "newPosts": "New posts from your community"
```
@LindasLinguas line 16 here is new 😊🙏🏼

LindasLinguas replied:
```suggestion
    "newPosts": "Neue Beiträge aus deiner Community"
```

LindasLinguas replied:
Community normally would translate to "Gemeinschaft", but in my eyes, this doesn't really fit in Journaly's context... That's why I kept the English word "Community".

robin-macpherson commented on packages/web/public/static/locales/de/notifications.json:
```diff
@@ -0,0 +1,22 @@
+  },
+  "levelTwo": {
+    "newFollowers": "Diese {{followerCount}} großartigen Journaler folgen dir nun",
+    "threadCommentThanks": "{{userIdentifier}} said thank you for the following {{thanksCount}} feedback comments",
```
@LindasLinguas "threadCommentThanks" has been modified

LindasLinguas replied:
```suggestion
    "threadCommentThanks": "{{userIdentifier}} hat für die folgendenden {{thanksCount}} Feedback-Kommentare Danke gesagt",
```

robin-macpherson commented:
@LindasLinguas good news! We now know how to implement singular vs. plural translations programmatically, and I've updated the translation files accordingly. 

For now, as I'm trying to get so many things done, I simply added your original translation to both the singular & plural keys, but please note that I also adjusted the original English ones a little bit, too, because now we can really tailor them and make it an even nicer experience for the user 🥰 It might be helpful to cross-reference the English file as you go through to update them.

So the file needing attention is `/de/notifications.json` and the file to reference is `/en/notifications.json`

Thanks so much!

robin-macpherson commented:
@LindasLinguas I also just added a few more lines where you will see the word `Subscribed` on the end of the key, for example `threadCommentsSubscribed_one`. This is to make it clear if the comments are on one of YOUR posts or someone else's post (the technical term is being "subscribed" to a comment thread).

So now if it is your post, it says "There is 1 new feedback comment on your post", but if it's on someone else's post it says "There is 1 new feedback comment on this post" (you will also be able to see the post image and the post title, but this is just to add a little extra element of clarity).

LindasLinguas commented on packages/web/public/static/locales/de/notifications.json:
```diff
@@ -0,0 +1,35 @@
+  "feedTitle": "Benachrichtigungen",
+  "emptyFeed": "Keine Benachrichtigungen ... bis jetzt!",
+  "levelOne": {
+    "newFollowers": "Du hast 1 neue Follower",
```
```suggestion
    "newFollowers": "Du hast einen neuen Follower",
```

LindasLinguas commented on packages/web/public/static/locales/de/notifications.json:
```diff
@@ -0,0 +1,35 @@
+  "levelOne": {
+    "newFollowers": "Du hast 1 neue Follower",
+    "newFollowers_plural": "Du hast {{count}} neue Follower",
+    "threadComments": "Du hast 1 neue Feedback-Kommentare zu diesem Beitrag",
```
```suggestion
    "threadComments": "Du hast einen neuen Feedback-Kommentar zu deinem Beitrag",
```

LindasLinguas commented on packages/web/public/static/locales/de/notifications.json:
```diff
@@ -0,0 +1,35 @@
+    "newFollowers": "Du hast 1 neue Follower",
+    "newFollowers_plural": "Du hast {{count}} neue Follower",
+    "threadComments": "Du hast 1 neue Feedback-Kommentare zu diesem Beitrag",
+    "threadComments_plural": "Du hast {{count}} neue Feedback-Kommentare zu diesem Beitrag",
```
```suggestion
    "threadComments_plural": "Du hast {{count}} neue Feedback-Kommentare zu deinem Beitrag",
```

LindasLinguas commented on packages/web/public/static/locales/de/notifications.json:
```diff
@@ -0,0 +1,35 @@
+    "newFollowers_plural": "Du hast {{count}} neue Follower",
+    "threadComments": "Du hast 1 neue Feedback-Kommentare zu diesem Beitrag",
+    "threadComments_plural": "Du hast {{count}} neue Feedback-Kommentare zu diesem Beitrag",
+    "threadCommentsSubscribed": "There is 1 new feedback comment on this post",
```
```suggestion
    "threadCommentsSubscribed": "Es gibt einen neuen Feedback-Kommentar zu diesem Beitrag",
```

LindasLinguas commented on packages/web/public/static/locales/de/notifications.json:
```diff
@@ -0,0 +1,35 @@
+    "threadComments": "Du hast 1 neue Feedback-Kommentare zu diesem Beitrag",
+    "threadComments_plural": "Du hast {{count}} neue Feedback-Kommentare zu diesem Beitrag",
+    "threadCommentsSubscribed": "There is 1 new feedback comment on this post",
+    "threadCommentsSubscribed_plural": "There are {{count}} new feedback comments on this post",
```
```suggestion
    "threadCommentsSubscribed_plural": "Es gibt {{count}} neue Feedback-Kommentare zu diesem Beitrag",
```

LindasLinguas commented on packages/web/public/static/locales/de/notifications.json:
```diff
@@ -0,0 +1,35 @@
+    "threadComments_plural": "Du hast {{count}} neue Feedback-Kommentare zu diesem Beitrag",
+    "threadCommentsSubscribed": "There is 1 new feedback comment on this post",
+    "threadCommentsSubscribed_plural": "There are {{count}} new feedback comments on this post",
+    "postComments": "Du hast 1 neue allgemeine Kommentare zu diesem Beitrag",
```
```suggestion
    "postComments": "Du hast einen neuen allgemeinen Kommentar zu deinem Beitrag",
```

LindasLinguas commented on packages/web/public/static/locales/de/notifications.json:
```diff
@@ -0,0 +1,35 @@
+    "threadCommentsSubscribed": "There is 1 new feedback comment on this post",
+    "threadCommentsSubscribed_plural": "There are {{count}} new feedback comments on this post",
+    "postComments": "Du hast 1 neue allgemeine Kommentare zu diesem Beitrag",
+    "postComments_plural": "Du hast {{count}} neue allgemeine Kommentare zu diesem Beitrag",
```
```suggestion
    "postComments_plural": "Du hast {{count}} neue allgemeine Kommentare zu deinem Beitrag",
```

LindasLinguas commented on packages/web/public/static/locales/de/notifications.json:
```diff
@@ -0,0 +1,35 @@
+    "threadCommentsSubscribed_plural": "There are {{count}} new feedback comments on this post",
+    "postComments": "Du hast 1 neue allgemeine Kommentare zu diesem Beitrag",
+    "postComments_plural": "Du hast {{count}} neue allgemeine Kommentare zu diesem Beitrag",
+    "postCommentsSubscribed": "There is 1 new general comment on this post",
```
```suggestion
    "postCommentsSubscribed": "Es gibt einen neuen allgemeinen Kommentar zu diesem Beitrag",
```

LindasLinguas commented on packages/web/public/static/locales/de/notifications.json:
```diff
@@ -0,0 +1,35 @@
+    "postComments": "Du hast 1 neue allgemeine Kommentare zu diesem Beitrag",
+    "postComments_plural": "Du hast {{count}} neue allgemeine Kommentare zu diesem Beitrag",
+    "postCommentsSubscribed": "There is 1 new general comment on this post",
+    "postCommentsSubscribed_plural": "There are {{count}} new general comments on this post",
```
```suggestion
    "postCommentsSubscribed_plural": "Es gibt {{count}} neue allgemeine Kommetare zu diesem Beitrag",
```

LindasLinguas commented on packages/web/public/static/locales/de/notifications.json:
```diff
@@ -0,0 +1,35 @@
+    "postComments_plural": "Du hast {{count}} neue allgemeine Kommentare zu diesem Beitrag",
+    "postCommentsSubscribed": "There is 1 new general comment on this post",
+    "postCommentsSubscribed_plural": "There are {{count}} new general comments on this post",
+    "postClaps": "1 Leute haben diesem Beitrag applaudiert",
```
```suggestion
    "postClaps": "Ein Journaler hat diesem Beitrag applaudiert",
```

LindasLinguas commented on packages/web/public/static/locales/de/notifications.json:
```diff
@@ -0,0 +1,35 @@
+    "postCommentsSubscribed": "There is 1 new general comment on this post",
+    "postCommentsSubscribed_plural": "There are {{count}} new general comments on this post",
+    "postClaps": "1 Leute haben diesem Beitrag applaudiert",
+    "postClaps_plural": "{{count}} Leute haben diesem Beitrag applaudiert",
```
```suggestion
    "postClaps_plural": "{{count}} Journaler haben diesem Beitrag applaudiert",
```

LindasLinguas commented on packages/web/public/static/locales/de/notifications.json:
```diff
@@ -0,0 +1,35 @@
+    "postCommentsSubscribed_plural": "There are {{count}} new general comments on this post",
+    "postClaps": "1 Leute haben diesem Beitrag applaudiert",
+    "postClaps_plural": "{{count}} Leute haben diesem Beitrag applaudiert",
+    "threadCommentThanks": "{{userIdentifier}} hat für 1 Feedback-Kommentare Danke gesagt",
```
```suggestion
    "threadCommentThanks": "{{userIdentifier}} hat für einen Feedback-Kommentar Danke gesagt",
```

LindasLinguas commented on packages/web/public/static/locales/de/notifications.json:
```diff
@@ -0,0 +1,35 @@
+    "postClaps_plural": "{{count}} Leute haben diesem Beitrag applaudiert",
+    "threadCommentThanks": "{{userIdentifier}} hat für 1 Feedback-Kommentare Danke gesagt",
+    "threadCommentThanks_plural": "{{userIdentifier}} hat für {{count}} Feedback-Kommentare Danke gesagt",
+    "newPosts": "Journaler, denen du folgst, haben 1 neue Beiträge veröffentlicht",
```
```suggestion
    "newPosts": "Journaler, denen du folgst, haben einen neuen Beitrag veröffentlicht",
```

LindasLinguas commented on packages/web/public/static/locales/de/notifications.json:
```diff
@@ -0,0 +1,35 @@
+    "newPosts_plural": "Journaler, denen du folgst, haben {{count}} neue Beiträge veröffentlicht"
+  },
+  "levelTwo": {
+    "newFollowers": "Diese 1 großartigen Journaler folgen dir nun",
```
```suggestion
    "newFollowers": "Dieser großartige Journaler folgt dir nun",
```

LindasLinguas commented on packages/web/public/static/locales/de/notifications.json:
```diff
@@ -0,0 +1,35 @@
+  "levelTwo": {
+    "newFollowers": "Diese 1 großartigen Journaler folgen dir nun",
+    "newFollowers_plural": "Diese {{count}} großartigen Journaler folgen dir nun",
+    "threadCommentThanks": "{{userIdentifier}} hat für die folgendenden 1 Feedback-Kommentare Danke gesagt",
```
```suggestion
    "threadCommentThanks": "{{userIdentifier}} hat für den folgenden Feedback-Kommentar Danke gesagt",
```

LindasLinguas commented on packages/web/public/static/locales/de/notifications.json:
```diff
@@ -0,0 +1,35 @@
+    "newFollowers": "Diese 1 großartigen Journaler folgen dir nun",
+    "newFollowers_plural": "Diese {{count}} großartigen Journaler folgen dir nun",
+    "threadCommentThanks": "{{userIdentifier}} hat für die folgendenden 1 Feedback-Kommentare Danke gesagt",
+    "threadCommentThanks_plural": "{{userIdentifier}} hat für die folgendenden {{count}} Feedback-Kommentare Danke gesagt",
```
```suggestion
    "threadCommentThanks_plural": "{{userIdentifier}} hat für die folgenden {{count}} Feedback-Kommentare Danke gesagt",
```

LindasLinguas commented on packages/web/public/static/locales/de/notifications.json:
```diff
@@ -0,0 +1,35 @@
+    "newFollowers_plural": "Diese {{count}} großartigen Journaler folgen dir nun",
+    "threadCommentThanks": "{{userIdentifier}} hat für die folgendenden 1 Feedback-Kommentare Danke gesagt",
+    "threadCommentThanks_plural": "{{userIdentifier}} hat für die folgendenden {{count}} Feedback-Kommentare Danke gesagt",
+    "postClaps": "Diese 1 Leute haben deinem Beitrag applaudiert",
```
```suggestion
    "postClaps": "Dieser Journaler hat deinem Beitrag applaudiert",
```

LindasLinguas commented on packages/web/public/static/locales/de/notifications.json:
```diff
@@ -0,0 +1,35 @@
+    "threadCommentThanks": "{{userIdentifier}} hat für die folgendenden 1 Feedback-Kommentare Danke gesagt",
+    "threadCommentThanks_plural": "{{userIdentifier}} hat für die folgendenden {{count}} Feedback-Kommentare Danke gesagt",
+    "postClaps": "Diese 1 Leute haben deinem Beitrag applaudiert",
+    "postClaps_plural": "Diese {{count}} Leute haben deinem Beitrag applaudiert",
```
```suggestion
    "postClaps_plural": "Diese {{count}} Journaler haben deinem Beitrag applaudiert",
```

LindasLinguas reviewed (commented):
Hi Robin, 

I see that you're constantly submitting stuff and I finally managed to do the missing German translations for the notification system! I'm so excited to see it implemented!

robin-macpherson commented:
Yay thank you @LindasLinguas ! 🙏🏼 Ahhhhh it has gotten so much better even since the preview you saw, I cannot waiiiiiiiiiit for you to see the finished work! We are planning to finish everything by the end of this Sunday 🤞🏼🐈🌟

LindasLinguas commented:
Hi Robin,

I was thinking about all the other translations that I want to review: Is it possible to put all translation sheets into an issue? So that I can review them the way I lately did?

Get Outlook for Android<https://aka.ms/ghei36>
________________________________
From: Robin MacPherson ***@***.***>
Sent: Friday, November 19, 2021 7:40:59 PM
To: Journaly/journaly ***@***.***>
Cc: LindasLinguas ***@***.***>; Mention ***@***.***>
Subject: Re: [Journaly/journaly] Build In-App Notifications System (#661)


Yay thank you @LindasLinguas<https://github.com/LindasLinguas> ! 🙏🏼 Ahhhhh it has gotten so much better even since the preview you saw, I cannot waiiiiiiiiiit for you to see the finished work! We are planning to finish everything by the end of this Sunday 🤞🏼🐈🌟

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/Journaly/journaly/pull/661#issuecomment-974315544>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AWA6MU2Z3EV7OOA6HX3WWXTUM2K3XANCNFSM5FGFGQIA>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

robin-macpherson commented:
> I was thinking about all the other translations that I want to review: Is it possible to put all translation sheets into an issue? So that I can review them the way I lately did?

@LindasLinguas it's possible but not as straight-forward because I would need to make a change to each of the files for them to show up in a Pull Request. Maybe I can show you sometime how you can make the changes and submit a Pull Request yourself

LindasLinguas commented:
> > I was thinking about all the other translations that I want to review: Is it possible to put all translation sheets into an issue? So that I can review them the way I lately did?
> 
> @LindasLinguas it's possible but not as straight-forward because I would need to make a change to each of the files for them to show up in a Pull Request. Maybe I can show you sometime how you can make the changes and submit a Pull Request yourself. But ideally we would get the Translation Tools working nicely since this was the exact purpose they were created 😊 We may need a bit of time for that, though.
___
I'd be also fine with creating my own pull request! But I don't want to create more work on your side doing so :D

robin-macpherson commented on packages/web/public/static/locales/de/my-posts.json:
```diff
@@ -3,5 +3,6 @@
   "publishedEmptyState": "Du hast noch keine Beiträge veröffentlicht. Du kannst entweder einen Entwurf veröffentlichen oder <1>einen neuen Post</1>.",
-  "draftEmptyState": "Du hast keine Entwürfe."
+  "draftEmptyState": "Du hast keine Entwürfe.",
+  "errorLoadingPostsMessage": "There was an error retrieving your posts."
```
Hi @LindasLinguas , I just added this translation that I found we weren't yet translating in the UI. I'm not sure if you'll be able to get to it in time as we are going to be finally deploying everything to production in the next couple of hours, but I thought I'd include it and tag you just in case!

LindasLinguas replied:
```suggestion
  "errorLoadingPostsMessage": "Beim Abrufen deiner Beiträge ist ein Fehler aufgetreten."
```

LindasLinguas commented on packages/web/public/static/locales/de/my-feed.json:
```diff
@@ -1,19 +1,6 @@
 {
   "pageTitle": "Mein Feed",
   "noPostsMessage": "Es gibt noch nichts zu sehen ...",
-  "clearFilters": "Filter löschen",
```
Why have these lines been deleted?
